File: matrix/matrix.java
class matrix
{
	public static void main(String args[]) {
		int size = 1;
		for (int i = 1; i <= 20; i++) {
			int a[][];
			int b[][];
			a = new int[size][size];
			b = new int[size][size];
			int c[][];
			c = new int[size][size];

			for (int j = 0; j < size; j++) {
				for (int k = 0; k < size; k++) {
					a[j][k] = 1;
					b[j][k] = 1;
					c[j][k] = 0;
				}
			}
			long tstart = System.nanoTime();
			for (int p = 0; p < size; p++) {
				for (int q = 0; q < size; q++) {
					for (int r = 0; r < size; r++) {
						c[p][q] += a[p][r] * b[r][q];
					}
				}
			}
			long tend = System.nanoTime();

			System.out.println((tend - tstart) / 1000000000f);
			size *= 2;
		}
	}
}

File: matrix/matrix.py
# -*- coding: utf-8 -*-
import time

def matrix(size):
    tstart = time.time()
    a = [[1 for i in range(size)] for j in range(size)]
    c = [[0 for i in range(size)] for j in range(size)]
    for i in range(size):
        for j in range(size):
            for k in range(size):
                c[i][j] += a[i][k] * a[k][j]
    tend = time.time()
    print (tend - tstart)


if __name__ == "__main__":
    size = 1
    for i in range(10):
        size *= 2
        matrix(size)
